### ISSUE TITLE
Exactly what does `DynamicImage.raw_pixels()` return?

### ISSUE THREAD

**Issue by brownjohnf** [image-rs/image#1011]
I'm attempting to run a checksum on a photo (JPEG in this case, but I'd like to extend it to other types), and I'd like the checksum to remain constant even if metadata changes. I found the `DynamicImage.raw_pixels` method, and it looked like what I needed, but it appears that its contents (a `Vec<u8>`) change when metadata is modified on the image.

Does it actually return the bytes for the whole image, and not just the bytes making up the pixel definition? Is there a way to get the raw contents of the image, without any of the metadata?

My use-case here is locating duplicate images which may have had different metadata applied to them over time by various photo management tools.

**Comment by fintelia** [comment]
That method only returns the bytes making up the image. However, the image data itself can be impacted by certain metadata changes. For instance if the image contains an alpha channel then there will be extra bytes in the image devoted to storing it (even if the whole image is still opaque). The JPEG compression format is also lossy, so it is possible that the images you have look identical to a human observer but actually have color values that differ by a fraction of a percent.

Of course it is also possible that there is just a bug in the implementation of that function. Could you elaborate on which metadata changes you are making and perhaps even share which identical images are producing different hashes?

**Comment by aengelas** [comment]
Thanks for such a helpful response! I was seeing the issues while running tests in my project where I added and removed tags, checking the checksum in between (so the image _should_ have stayed the same). I went back and and found that I might have been causing the problem in my test suite. Cleaning things up to be less coupled appears to have resolved the problem, though I still don't understand what I was doing to cause the problem.

I think I'll close this for now, since I can't replicate with a correctly sandboxed environment. I suspect the error was all on my end. Thanks again!
